Monika After Story/game/script-affection.rpy

```diff
@@ -144,7 +145,16 @@ init python:
         ):
         if not persistent._mas_affection_badexp_freeze:
             #Otherwise, use the value passed in the argument.
-            persistent._mas_affection["affection"] += amount
+            persistent._mas_affection["affection"] -= amount
+            #Updates the experience levels if necessary.
+            mas_updateAffectionExp()
+
+    def mas_setAffection(
+            amount=persistent._mas_affection["affection"]
+        ):
+        if not persistent._mas_affection_badexp_freeze or persistent_mas_affection_goodexp_freeze:
```

Comment:
This condition translates to:
`if badexp_freeze is False or goodexp_freeze is True`
If you mean to freeze affection if either of these values are true, then you'd want it like:
`if not badexp_freeze and not goodexp_freeze`
Another way to do this is:
`if not (badexp_freeze or goodexp_freeze)`